paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveMetastoreClient.java

```diff
@@ -105,55 +85,56 @@ public void addPartition(LinkedHashMap<String, String> partitionSpec) throws Exc
     }
 
     @Override
-    public void addPartitionsSpec(List<LinkedHashMap<String, String>> partitionSpecsList)
-            throws Exception {
+    public void addPartitions(List<LinkedHashMap<String, String>> partitions) throws Exception {
         int currentTime = (int) (System.currentTimeMillis() / 1000);
         List<Partition> hivePartitions =
-                partitionSpecsList.stream()
+                partitions.stream()
                         .map(partitionSpec -> toHivePartition(partitionSpec, currentTime))
                         .collect(Collectors.toList());
         clients.execute(client -> client.add_partitions(hivePartitions, true, false));
     }
 
     @Override
     public void alterPartition(
-            LinkedHashMap<String, String> partitionSpec,
-            Map<String, String> parameters,
-            long modifyTime,
-            boolean ignoreIfNotExist)
+            LinkedHashMap<String, String> partition, PartitionStats partitionStats)
             throws Exception {
-        List<String> partitionValues = new ArrayList<>(partitionSpec.values());
-        int currentTime = (int) (modifyTime / 1000);
-        Partition hivePartition;
+        List<String> partitionValues = new ArrayList<>(partition.values());
+
+        Map<String, String> statistic = new HashMap<>();
```

Comment:
Map<String, String> statistic = new HashMap<>(5);

Reply:
For these small collection, we don't need to specific the init size.